packages/shared/lib/core/wallet/utils/outputs/getExpirationDateFromOutput.ts

```diff
@@ -0,0 +1,17 @@
+import { OutputTypes } from '@iota/types'
+import { MILLISECONDS_PER_SECOND } from '@lib/time'
+import { OUTPUT_TYPE_TREASURY, UNLOCK_CONDITION_EXPIRATION } from '../../constants'
+
+export function getExpirationDateFromOutput(output: OutputTypes): Date {
+    if (output.type !== OUTPUT_TYPE_TREASURY) {
```

Comment:
Would it be easier to have an `OutputType` enum where we use the actual corresponding protocol numbers rather than managing multiple constants?

Reply:
We tried this but it doesn't allow for type narrowing and neither did an object literal so we had no other choice.

Reply:
Thats the problem i wrote on discord. Using enums does not narrow the type